src/class_loader_core.cpp

```diff
@@ -445,31 +443,32 @@ void loadLibrary(const std::string & library_path, ClassLoader * loader)
     return;
   }
 
-  Poco::SharedLibrary * library_handle = nullptr;
+  #ifdef WIN32
+  HINSTANCE library_handle;
+  #else
+  void * library_handle = nullptr;
+  #endif
   static std::recursive_mutex loader_mutex;
 
   {
     std::lock_guard<std::recursive_mutex> loader_lock(loader_mutex);
 
-    try {
-      setCurrentlyActiveClassLoader(loader);
-      setCurrentlyLoadingLibraryName(library_path);
-      library_handle = new Poco::SharedLibrary(library_path);
-    } catch (const Poco::LibraryLoadException & e) {
-      setCurrentlyLoadingLibraryName("");
-      setCurrentlyActiveClassLoader(nullptr);
-      throw class_loader::LibraryLoadException(
-              "Could not load library (Poco exception = " + std::string(e.message()) + ")");
-    } catch (const Poco::LibraryAlreadyLoadedException & e) {
+    setCurrentlyActiveClassLoader(loader);
+    setCurrentlyLoadingLibraryName(library_path);
+    #ifdef WIN32
+    library_handle = LoadLibrary(library_path.c_str());
+    #else
+    library_handle = dlopen(library_path.c_str(), RTLD_LAZY);
+    #endif
+    if (!library_handle) {
       setCurrentlyLoadingLibraryName("");
       setCurrentlyActiveClassLoader(nullptr);
+      #ifdef WIN32
+      throw class_loader::LibraryLoadException("Could not load library");
+      #else
       throw class_loader::LibraryLoadException(
-              "Library already loaded (Poco exception = " + std::string(e.message()) + ")");
-    } catch (const Poco::NotFoundException & e) {
-      setCurrentlyLoadingLibraryName("");
-      setCurrentlyActiveClassLoader(nullptr);
-      throw class_loader::LibraryLoadException(
-              "Library not found (Poco exception = " + std::string(e.message()) + ")");
+              "Could not load library (" + std::string(dlerror()) + ")");
+      #endif
```

Comment:
This throwing logic should also be encapsulated into a function or as part of the "load library" function I asked you to factor into above.